core/src/main/java/com/amplifyframework/core/model/ModelSchema.java

```diff
@@ -161,6 +162,27 @@ public static ModelSchema fromModelClass(@NonNull Class<? extends Model> clazz)
         }
     }
 
+    /**
+     * Returns the applicable auth rules for a given operation.
+     * @param modelOperation The desired operation type (read, create, update, delete).
+     * @return A list of {@link AuthRule}s for the given operation.
+     */
+    public List<AuthRule> getApplicableRules(ModelOperation modelOperation) {
```

Comment:
I initially was returning just a list of auth types, but there are instances where we need to know whether userPools is being used with an owner rule or private rule for example.

For example, when establishing subscriptions, if the only READ permission is tied to an owner-based rule, we have to decorate the subscription request body with the owner. But, if there is an auth rule that allows READs with the private strategy, then I will still need to use userPools, but won't decorate the request body with the ownerId.

Regarding the field-level use case, that's definitely a gap I have to close to take into account the default auth type if no auth rule is specified at the object level.

Reply:
Quick follow up on the 

> Hmm, like you, I'm a bit conflicted about whether to use an enum or String to represent the auth strategy. I think I'm leaning more towards representing it as an enum, because that's how it is defined on the CLI side (See [@auth rule definition](https://docs.amplify.aws/cli/graphql-transformer/auth#definition)).
> 
> I do share your concern though that it _sort of_ has some AWS specific concepts (`AWS_IAM`, `AMAZON_COGNITO_USER_POOLS`, etc), which would be leaking into core. I think they could be generalized though, by renaming the enum case values (e.g. just `IAM`, and `USER_POOLS`). Basically, we could match the naming that exists in the CLI definition of the auth directive. After making it more general, I think it would be okay to move `AuthorizationType` into `core`. What do you think?

Quick follow up on this. The values that the CLI puts in the config file reflect the values used by the [AppSync cloudformation resource](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-appsync-graphqlapi.html#cfn-appsync-graphqlapi-authenticationtype), which match the values we have now. With that said, I'm going to leave those names as is for now.